pkg/daemon/daemon.go

```diff
@@ -93,398 +55,304 @@ func New(
-		log.Log.V(2).Info("unsupported SriovOperatorConfig", "namespace", newCfg.Namespace, "name", newCfg.Name)
-		return
+func (dn *DaemonReconcile) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	reqLogger := log.FromContext(ctx).WithName("Reconcile")
+	// Get the latest NodeState
+	desiredNodeState := &sriovnetworkv1.SriovNetworkNodeState{}
+	err := dn.client.Get(ctx, client.ObjectKey{Namespace: req.Namespace, Name: req.Name}, desiredNodeState)
+	if err != nil {
+		if errors.IsNotFound(err) {
+			reqLogger.Info("NodeState doesn't exist")
+			return ctrl.Result{}, nil
+		}
+		reqLogger.Error(err, "Failed to fetch node state", "name", vars.NodeName)
+		return ctrl.Result{}, err
 	}
 
-	snolog.SetLogLevel(newCfg.Spec.LogLevel)
-
-	newDisableDrain := newCfg.Spec.DisableDrain
-	if dn.disableDrain != newDisableDrain {
-		dn.disableDrain = newDisableDrain
-		log.Log.Info("Set Disable Drain", "value", dn.disableDrain)
+	// Check the object as the drain controller annotations
+	// if not just wait for the drain controller to add them before we start taking care of the nodeState
+	if !utils.ObjectHasAnnotationKey(desiredNodeState, consts.NodeStateDrainAnnotationCurrent) ||
+		!utils.ObjectHasAnnotationKey(desiredNodeState, consts.NodeStateDrainAnnotation) {
+		reqLogger.V(2).Info("NodeState doesn't have the current drain annotation")
+		return ctrl.Result{}, nil
 	}
 
-	if !reflect.DeepEqual(oldCfg.Spec.FeatureGates, newCfg.Spec.FeatureGates) {
-		dn.featureGate.Init(newCfg.Spec.FeatureGates)
-		log.Log.Info("Updated featureGates", "featureGates", dn.featureGate.String())
-	}
+	latest := desiredNodeState.GetGeneration()
+	current := desiredNodeState.DeepCopy()
+	reqLogger.V(0).Info("new generation", "generation", latest)
 
-	vars.MlxPluginFwReset = dn.featureGate.IsEnabled(consts.MellanoxFirmwareResetFeatureGate)
-}
+	// Update the nodeState Status object with the existing network state (interfaces bridges and rdma status)
+	err = dn.getHostNetworkStatus(desiredNodeState)
+	if err != nil {
+		reqLogger.Error(err, "failed to get host network status")
+		return ctrl.Result{}, err
+	}
 
-func (dn *Daemon) nodeStateSyncHandler() error {
-	var err error
-	// Get the latest NodeState
-	var sriovResult = &systemd.SriovResult{SyncStatus: consts.SyncStatusSucceeded, LastSyncError: ""}
-	dn.desiredNodeState, err = dn.sriovClient.SriovnetworkV1().SriovNetworkNodeStates(vars.Namespace).Get(context.Background(), vars.NodeName, metav1.GetOptions{})
+	// if we are running in systemd mode we want to get the sriov result from the config-daemon that runs in systemd
+	sriovResult, exist, err := dn.checkSystemdStatus(ctx, desiredNodeState)
+	//TODO: in the case we need to think what to do if we try to apply again or not
 	if err != nil {
-		log.Log.Error(err, "nodeStateSyncHandler(): Failed to fetch node state", "name", vars.NodeName)
-		return err
+		reqLogger.Error(err, "failed to check systemd status unexpected error")
+		return ctrl.Result{}, nil
```

Comment:
we dont want to fail here ?

Reply:
If we return an error it will get stuck in a loop as if the result for the systemd operator return an error we don't retry to not put the node into a bootloop

Reply:
ack.

 should we update status with "unexpected error" ?

Reply:
right nice catch!